pkgs/applications/networking/instant-messengers/webcord/webcord-vencord/vencord-web-extension/default.nix

```diff
@@ -0,0 +1,44 @@
+{ buildNpmPackage
+, fetchFromGitHub
+, lib
+, substituteAll
+,
+}:
+buildNpmPackage rec {
+  pname = "vencord-web-extension";
+  version = "1.1.6";
+
+  src = fetchFromGitHub {
+    owner = "Vendicated";
+    repo = "Vencord";
+    rev = "v${version}";
+    sha256 = "sha256-V9fzSoRqVlk9QqpzzR2x+aOwGHhQhQiSjXZWMC0uLnQ=";
+  };
+
+  npmDepsHash = "sha256-jKSdeyQ8oHw7ZGby0XzDg4O8mtH276ykVuBcw7dU/Ls=";
+  npmFlags = [ "--legacy-peer-deps" ];
+  npmBuildScript = "buildWeb";
+
+  prePatch = ''
+    cp ${./package-lock.json} ./package-lock.json
+  '';
+
+  patches = [
+    (substituteAll {
+      src = ./replace-git.patch;
+      inherit version;
+    })
+  ];
+
+  installPhase = ''
+    cp -r dist/extension-unpacked $out
+  '';
+
+  meta = with lib; {
+    description = "Vencord web extension";
+    homepage = "https://github.com/Vendicated/Vencord";
+    platforms = [ "x86_64-linux" ];
```

Comment:
Couldn't get it to build vencord in arm64. It tries downloading "esbuild-linux-arm64" while building instead of at the vendor derivation.

I figured I'll mark it as x86_64 only for now and get this merged. arm support can be figured out later.

Reply:
Set `ESBUILD_BINARY_PATH` to `lib.getExe esbuild`.